src/common.rs

```diff
@@ -37,6 +37,18 @@ impl ColorType {
             _ => None,
         }
     }
+
+    pub(crate) fn raw_row_length_from_width(&self, depth: BitDepth, width: u32) -> usize {
+        let bits = width as usize * self.samples() * depth as usize;
```

Comment:
Does this need to use `u64` instead of `usize` to prevent possible overflow on 32-bit platforms?

Reply:
Tbh we should catch this overflow __way__ earlier if anything. Do you have some test cases in mind, since this mainly tries to address the single issue reported with mismatching color/depth between different size calculations and not much else.

Reply:
What I'm worried about would be something like an 8bpp 600M x 1px image gray scale image. That would only take ~1GB so it should be a valid PNG and pass the other checks. However, when we compute the number of _bits_ required to store a row we could get an overflow.

Reply:
We could probably test that if we use the procedure outlined in #224 of ensuring that everything is computed in advance. Then a test image containing only the header and no IDAT chunk would be enough to check for the early decoding error. We need to check early against the limits in any case as well!

Reply:
I'm not sure I follow. We shouldn't be returning any error in this case, the function should just correctly return the number of bytes required to store a single row (which will always be less than usize::MAX). The issue is that intermediate computation needs a few extra bits of precision (35 vs 32). The fix should be as simple as writing some `as u64` casts on this line, and then cast back to a `usize` after doing the division on the next line.

Reply:
Seems like a good idea. Let's fork that in a separate issue, okay?